core/tests/performance_framework/perf_domain.py

```diff
@@ -134,13 +156,107 @@ def _get_duration_millisecs(self, event_end, event_initial):
         return end_timestamp - initial_timestamp
 
     def get_page_load_time_millisecs(self):
-        """Returns the total page load time (in milliseconds)."""
+        """Returns the total page load time."""
         return self._get_duration_millisecs('loadEventEnd', 'fetchStart')
 
     def get_dom_ready_time_millisecs(self):
-        """Returns the total dom ready time (in milliseconds)."""
+        """Returns the time spent constructing the dom tree."""
         return self._get_duration_millisecs('domComplete', 'domInteractive')
 
     def get_request_time_millisecs(self):
-        """Returns the total request time (in milliseconds)."""
+        """Returns the time spent during request."""
         return self._get_duration_millisecs('responseEnd', 'requestStart')
+
+    def get_ready_start_time_millisecs(self):
+        """Returns the time consumed preparing the new page."""
+        return self._get_duration_millisecs('fetchStart', 'navigationStart')
+
+    def get_redirect_time_millisecs(self):
+        """Returns the time spent during redirection."""
+        return self._get_duration_millisecs('redirectEnd', 'redirectStart')
+
+    def get_appcache_time_millisecs(self):
+        """Returns the time spent for appcache."""
+        return self._get_duration_millisecs('domainLookupStart', 'fetchStart')
+
+    def get_unload_event_time_millisecs(self):
+        """Returns the time spent unloading documents."""
+        return self._get_duration_millisecs(
+            'unloadEventEnd', 'unloadEventStart')
+
+    def get_lookup_domain_time_millisecs(self):
+        """Returns the time spent for the domain name lookup for the current
+        document."""
+        return self._get_duration_millisecs(
+            'domainLookupEnd', 'domainLookupStart')
+
+    def get_connect_time_millisecs(self):
+        """Returns the time spent for establishing the connection to the server
+        to retrieve the current document."""
+        return self._get_duration_millisecs('connectEnd', 'connectStart')
+
+    def get_init_dom_tree_time_millisecs(self):
+        """Returns the time for request to completion of DOM loading."""
+        return self._get_duration_millisecs('domInteractive', 'responseEnd')
+
+    def get_load_event_time_millisecs(self):
+        """Returns the time spent for completion of the load event of the
+        current document. The load event is fired when a resource and its
+        dependent resources have finished loading."""
+        return self._get_duration_millisecs('loadEventEnd', 'loadEventStart')
+
+    def print_details(self):
+        """Helper function to print details for all the events."""
+        if not self.page_session_timings:
+            return
```

Comment:
@BenHenning  Yup. Also, should we call it in  `__init__` by default. Since, it will only be printed if the tests fails. And, if some tests fails we'll be better able to pin point the exact cause?